File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int n = nums.length;
        int cnt = 0;
        int ans = 0;
        int s = 0; int e = 0;
        while( e < n){
            if(nums[e] == 0) cnt ++;

            while(s <= e && cnt > k){
                if(nums[s] == 0) cnt--;
                s++;
            }
            ans = Math.max(ans, e - s + 1);
            e++;
        }
        return ans;
    }
}

File: LeetCode/Medium/1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int start = 0;
        int end = 0;
        int z = 0;
        int ans = 0;
        while(end < nums.size()){
            if(nums[end] == 0) z++;
            while(z > k){
                if(nums[start] == 0) z--;
                start++;
            }
            ans = max(ans, end - start + 1);
            end++;
        }
        return ans;
    }
};